File: Models/Domain/Walk.cs
namespace NZWalksAPI.Models.Domain
{
    public class Walk
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public required double Length { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }

        public Guid DifficultyId { get; set; }
        public Guid RegionId { get; set; }

        // Navigation property:
        public required Difficulty Difficulty { get; set; }
        public required Region Region { get; set; }
    }
}

File: frontend/src/data/walk.ts
// import { IRegion } from "./region";

export interface IWalk {
	id: string;
	name: string;
	length: number;
	description?: string;
	imageUrl?: string;
	// region?: IRegion;
	// difficulty?: object
}

export const WalkHeaders = {
	id: "ID",
	name: "Name",
	Length: "Length",
	// description: "Description",
	imageUrl: "Image",
	// region: "Region",
	// difficulty: "Difficulty",
};
